### PR TITLE
Using new Declarative Jenkinsfile format

### PR DESCRIPTION
**Pull request by joakime**
+ Makes things easier to maintain.
+ Does not solve for bad "unstable" flagging of multiple
  stages when only one is failing.
  See: https://issues.jenkins-ci.org/browse/JENKINS-39203
  But this Declarative format will help once that bug is fixed.

Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,95 +1,97 @@
...
-        withMaven(
-            maven: mvnName,
-            jdk: "$jdk",
-            publisherStrategy: 'EXPLICIT',
-            globalMavenSettingsConfig: settingsName,
-            mavenOpts: mavenOpts,
-            mavenLocalRepo: localRepo) {
-          sh "mvn -f aggregates/jetty-all-compact3 -V -B -Pcompact3 clean install -T6"
+        stage("Build Compact3") {
+          agent { node { label 'linux' } }
```
Comment (olamy) [inline Jenkinsfile]: here it can be executed on an other node

File: Jenkinsfile [hunk 2]
```diff
@@ -1,95 +1,97 @@
...
+            warnings consoleParsers: [[parserName: 'Maven'], [parserName: 'Java']]
+          }
         }
 
-        // Report on Maven and Javadoc warnings
-        step([$class        : 'WarningsPublisher',
-              consoleParsers: consoleParsers])
-      }
+        stage("Build Javadoc") {
+          agent { node { label 'linux' } }
```
[the remark this answers is not in the document]
Comment (joakime) [reply Jenkinsfile]: Oh yeah.

Meh, I'd rather have it build it all (but skip tests) on a different node ...

``` 
mavenBuild("jdk8", "clean install javadoc:javadoc -PskipTests")
```

File: Jenkinsfile (hunk 1 above)
Comment (joakime) [reply Jenkinsfile]: Ditto, same comment as above.

File: Jenkinsfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (olamy) [reply Jenkinsfile]: yup sure good idea worth to try that! might be faster build (but need more nodes :) )

**Comment by joakime** [comment]
Once 9.4.14 is released, I'll merge this into the `jetty-9.4.x` branch.

**Pull request merged**